packages/wallet/src/PersonalWallet/PersonalWallet.ts

```diff
@@ -137,19 +138,32 @@ export const DEFAULT_LOOK_AHEAD_SEARCH = 20;
 // Ideally we should calculate this based on the activeSlotsCoeff and probability of a single block per epoch.
 const BLOCK_SLOT_GAP_MULTIPLIER = 2.5;
 
-const isOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx): input is OutgoingTx =>
+const getSerializedTxCBOR = (input: TxCBOR) => TxCBOR.deserialize(input).body;
+const isOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is OutgoingTx =>
   typeof input === 'object' && 'cbor' in input;
-const isTxCBOR = (input: Cardano.Tx | TxCBOR | OutgoingTx): input is TxCBOR => typeof input === 'string';
-const processOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx): OutgoingTx => {
+const isTxCBOR = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is TxCBOR => typeof input === 'string';
+const isSubmitTxArgs = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): input is SubmitTxArgs =>
+  typeof input === 'object' && 'context' in input;
+const processOutgoingTx = (input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs): OutgoingTx => {
   // TxCbor
   if (isTxCBOR(input)) {
     return {
-      body: TxCBOR.deserialize(input).body,
+      body: getSerializedTxCBOR(input),
       cbor: input,
       // Do not re-serialize transaction body to compute transaction id
       id: Cardano.TransactionId.fromTxBodyCbor(TxBodyCBOR.fromTxCBOR(input))
     };
   }
+  // SubmitTxArgs
+  if (isSubmitTxArgs(input)) {
+    return {
+      body: input.signedTransaction.body,
```

Comment:
This shouldn't be working as `signedTransaction` should be an opaque string here :thinking: 
```suggestion
   const deserializedTx = TxCBOR.deserialize(input.signedTransaction);
   ...
      body: deserializedTx.body,
```

Reply:
2d88f9ab41c0bf941b8126409e8d597fa2713303  should address this